app/controllers/DevelopersController.scala

```diff
@@ -16,29 +16,65 @@
 
 package controllers
 
-import connectors.{AuthConnector, DeveloperConnector}
+import connectors.{ApiDefinitionConnector, ApplicationConnector, AuthConnector, DeveloperConnector}
 import model._
-import play.api.mvc.{Action, AnyContent}
+import model.Forms._
 import uk.gov.hmrc.play.frontend.controller.FrontendController
 import utils.{GatekeeperAuthProvider, GatekeeperAuthWrapper}
 import views.html.developers.developers
 
+import scala.concurrent.Future
+
 object DevelopersController extends DevelopersController {
   override val developerConnector: DeveloperConnector = DeveloperConnector
+  override val apiDefinitionConnector = ApiDefinitionConnector
+  override val applicationConnector = ApplicationConnector
+
   override def authConnector = AuthConnector
+
   override def authProvider = GatekeeperAuthProvider
 }
 
+
 trait DevelopersController extends FrontendController with GatekeeperAuthWrapper {
 
   val developerConnector: DeveloperConnector
+  val apiDefinitionConnector: ApiDefinitionConnector
+  val applicationConnector: ApplicationConnector
+
+  private def innerJoin(devs: Seq[User], apps: Seq[ApplicationResponse]): Seq[User] = {
+    val collaborators = apps.flatMap(_.collaborators).map(_.emailAddress).toSet
+    devs.filter(u => collaborators.contains(u.email))
+  }
```

Comment:
Fetching all developer emails + names for all apps that subscribe to API Y (or all applications, for that matter) should be a service concern IMO.  

Controllers should ideally be as smarts-free as possible, except for where that's concerned with dealing with HTTP/view-rendering specific logic.